### PR TITLE
feat(dropdownmenu): adding clickAwayListenerProps to manage clickAway behavior

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**DropdownMenu**
Github issue: #403 

The latest feature added to the Dropdown component is the ability to customize its click-away behavior using the `ClickAwayListenerProps` prop. This new prop allows passing through the Mouse and Pointer event handling props to the underlying Material UI `ClickAwayListener` component. By setting the mouseEvent and touchEvent props to false, the Dropdown menu can remain open even when users click outside of it.

To use this feature, include `ClickAwayListenerProps` as part of the `DropdownMenuProps`. For example, the following code shows how to create a Dropdown with a customized click-away behavior:

```javascript
<Dropdown
  label="Click Target"
  onChange={() => {}}
  options={GITHUB_LABELS}
  DropdownMenuProps={{
    ClickAwayListenerProps: {
      mouseEvent: false,
      touchEvent: false,
    },
  }}
/>
```

With this implementation, the Dropdown menu will remain open when users click outside of it using both mouse and touch events.

The `DropdownMenu` component can also achieve the same customized click-away behavior. For instance, you can use the `ClickAwayListenerProps` prop to disable the default behavior of closing the menu when users click outside of it.

Here's an example of a `DropdownMenu` component with a customized click-away behavior:

```javascript
<DropdownMenu
  multiple={multiple}
  onChange={handleChange}
  options={options}
  search={search}
  title={title}
  ClickAwayListenerProps={{
    mouseEvent: false,
    touchEvent: false,
  }}
  {...props}
/>

```

With the above code, the DropdownMenu will remain open even when users click outside of it using both mouse and touch events, thanks to the `mouseEvent` and `touchEvent` props set to false in the `ClickAwayListenerProps` prop.

The `mouseEvent` prop of the `clickAwayListenerProps` can be set to one of the following values:

- `onClick`
- `onMouseDown`
- `onMouseUp`
- `onPointerDown`
- `onPointerUp`
- `false`

The default value for mouseEvent is `onClick`.

Similarly, the `touchEvent` prop of the `ClickAwayListenerProps` can be set to one of the following values:

- `onTouchEnd`
- `onTouchStart`
- `false`

The default value for touchEvent is `onTouchEnd`.

More information about these props can be found here: https://mui.com/base/api/click-away-listener/

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [x] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

File: src/core/DropdownMenu/index.tsx [hunk 1]
```diff
@@ -67,6 +68,7 @@ interface ExtraDropdownMenuProps extends StyleProps {
   PaperComponent?: typeof StyledPaper | RenderFunctionType;
   children?: JSX.Element | null;
   onClickAway: (event: MouseEvent | TouchEvent) => void;
+  clickAwayListenerProps?: Partial<ClickAwayListenerProps>;
```
Comment (tihuan) [inline src/core/DropdownMenu/index.tsx:71]: nit: Can we use capital `C` instead to signify it's a component prop please 🙏 ? Following what MUI does and other props we have! Thanks so much!

```suggestion
  ClickAwayListenerProps?: Partial<ClickAwayListenerProps>;
```

**Review by tihuan — APPROVED** [review]
One tiny change to prop name otherwise LGTM!

File: src/core/DropdownMenu/index.tsx (hunk 1 above)
Comment (masoudmanson) [reply src/core/DropdownMenu/index.tsx:71]: Aah, for sure. Thanks for spotting this. 🥳

Comment (masoudmanson) [reply src/core/DropdownMenu/index.tsx:71]: It appears that there is already a ClickAwayListenerProps defined in MUI and is a reserved name. Therefore, we need to come up with a different name for our prop. Would it make sense to rename it to SdsClickAwayListenerProps instead? @tihuan

**Comment by cathystoli** [comment]
Release must have checklist:
- [ ] Zeroheight minimum documentation
- [ ] QA passing

Should haves:
- [ ] Figma component for DropDownmenu
- [ ] Component recipes

**Pull request merged**